ykman/cli/__main__.py

```diff
@@ -111,27 +111,27 @@ def _run_cmd_for_single(ctx, cmd, transports, reader=None):
             if len(readers) == 1:
                 return YubiKey(Descriptor.from_driver(readers[0]), readers[0])
             elif len(readers) > 1:
-                ctx.fail('Multiple YubiKeys on external readers detected.')
+                cli_fail('Multiple YubiKeys on external readers detected.')
             else:
-                ctx.fail('No YubiKey found on external reader.')
+                cli_fail('No YubiKey found on external reader.')
         else:
-            ctx.fail('Not a CCID command.')
+            cli_fail('Not a CCID command.')
     try:
         descriptors = get_descriptors()
     except usb.core.NoBackendError:
-        ctx.fail('No PyUSB backend detected!')
+        cli_fail('No PyUSB backend detected!')
     n_keys = len(descriptors)
     if n_keys == 0:
-        ctx.fail('No YubiKey detected!')
+        cli_fail('No YubiKey detected!')
     if n_keys > 1:
-        ctx.fail('Multiple YubiKeys detected. Use --device SERIAL to specify '
+        cli_fail('Multiple YubiKeys detected. Use --device SERIAL to specify '
                  'which one to use.')
     descriptor = descriptors[0]
     if descriptor.mode.transports & transports:
         try:
             return descriptor.open_device(transports)
         except FailedOpeningDeviceException:
-            ctx.fail('Failed connecting to {} [{}]. Make sure the application have \
+            cli_fail('Failed connecting to {} [{}]. Make sure the application have \
```

Comment:
Unrelated, but typo:
```suggestion
            cli_fail('Failed connecting to {} [{}]. Make sure the application has \
```